integration_tests/src/main/python/aqe_test.py

```diff
@@ -298,3 +298,40 @@ def do_it(spark):
 
     assert_gpu_and_cpu_are_equal_collect(do_it, conf=bhj_disable_conf)
 
+
+# See https://github.com/NVIDIA/spark-rapids/issues/10645. Sometimes the exchange can provide multiple
+# batches, so we to coalesce them into a single batch for the broadcast hash join.
+@ignore_order(local=True)
+@pytest.mark.skipif(not (is_databricks_runtime()), \
+    reason="Executor side broadcast only supported on Databricks")
+def test_aqe_join_executor_broadcast_enforce_single_batch():
+    # Use a small batch to see if Databricks could send multiple batches
+    conf = copy_and_update(_adaptive_conf, { "spark.rapids.sql.batchSizeBytes": "25" })
+    def prep(spark):
+        id_gen = RepeatSeqGen(IntegerGen(nullable=False), length=250)
+        name_gen = RepeatSeqGen(["Adam", "Bob", "Cathy"], data_type=StringType())
+        school_gen = RepeatSeqGen(["School1", "School2", "School3"], data_type=StringType())
+
+        df = gen_df(spark, StructGen([('id', id_gen), ('name', name_gen)], nullable=False), length=1000)
+        df.createOrReplaceTempView("df")
+
+        df_school = gen_df(spark, StructGen([('id', id_gen), ('school', school_gen)], nullable=False), length=250)
+        df.createOrReplaceTempView("df_school")
+
+    with_cpu_session(prep)
+
+    def do_it(spark):
+        res = spark.sql(
+            """
+                select /*+ BROADCAST(df_school) */ * from df, df_school where df.id == df_school.id
+            """
```

Comment:
nit: PySpark API looks nicer and would not require resorting to view registration along the lines

```Python
df.join(broadcast(df_school), df.id == df_school.id) 
```

Reply:
Understood, but this test requires a join hint to ensure a broadcast hash join is used in this test.

Reply:
I think my suggestion does that: https://github.com/apache/spark/blob/de2079163270d371e0ead33f09d61d66ee496e88/python/pyspark/sql/connect/functions/builtin.py#L317-L325